src/prefect/environments/storage/memory.py

```diff
@@ -0,0 +1,96 @@
+from typing import Any, Dict, Iterable, List, TYPE_CHECKING, Union
+
+import prefect
+from prefect.environments.storage import Storage
+
+if TYPE_CHECKING:
+    from prefect.core.flow import Flow
+    from prefect.engine.flow_runner import FlowRunner
+
+
+class Memory(Storage):
```

Comment:
At first I didn't understand how this worked but upon seeing the `build` function it became clear. Nice! May want to update the class's main docstring because it wasn't that obvious as to what it meant by `Memory`. (also just could be because I haven't had a coffee yet this morning)